enterprise/backend/src/metabase_enterprise/sandbox/query_processor/middleware/row_level_restrictions.clj

```diff
@@ -333,7 +330,9 @@
       original-query
       (-> sandboxed-query
           (assoc ::original-metadata (expected-cols original-query))
-          (update-in [::qp.perms/perms :gtaps] (fn [perms] (into (set perms) (sandboxes->perms-set (vals table-id->gtap)))))))))
+          (update-in [::query-perms/perms :gtaps]
+                     (fn [required-perms] (merge required-perms
```

Comment:
If I understand correctly, the parameter here should be something like `existing-perms` or `perms`? We're figuring out what additional perms we need for the sandboxed query to run, and adding them to the set of permissions the user already has - right?